### PR TITLE
feat: 75 drive status indicator local cloud available offline

### PR DESCRIPTION
**Pull request by ryanwolhuter**
NOTE: The huge number of changes are because I had to regenerate the lockfile. There was some kind of naming conflict with the react-aria-components library. Upgrading to the latest version fixed it.

Add a `StatusIndicator` component to be shown in the `secondaryIcon` slot of a drive in the tree view. Currently there is no icon shown there.

I have made the component handle all of the cases as specified here https://www.notion.so/makerdao-ses/CDN-1-08-Drive-status-indicator-local-cloud-available-offline-da1e6140c280476f88ce43e3e63188e1

But there are a couple of things in that specification that don't line up exactly with the existing code. 

1. There is no `error` state in the tree items, but the design calls for one. I added an optional `error` field that can hold an `Error` and be passed down as a prop to determine if the drive is in an error state.

2. The existing `ItemStatus` definition looks like this:

```
export enum ItemStatus {
    Available = 'available',
    AvailableOffline = 'available-offline',
    Syncing = 'syncing',
    Offline = 'offline',
}
```

This makes the statuses of "syncing" and "available-offline" mutually exclusive. But in the spec, we need to have different syncing states for items that are available offline vs cloud only. For now I have made the `StatusIndicator` component able to handle this distinction, and have left the translation work to the `DriveView` for now. Since we are already refactoring the `DriveView`, I did not want to do a bunch of redundant work that will be deleted in the refactor. The `StatusIndicator` component will work as-is when the refactor is ready.

I have added more drivevs to the `DriveView` stories that demonstrate each of the states described in the spec.

One thing I have noticed is that having these status indicators for all of the files and folders and drives makes the UI look a bit too busy. I think we need to consider showing less information. In the current designs, I notice that the status indicators for the individual files and folders are absent.

Also the screenshots in the spec show the status icons as being quite a lot smaller than the ones used for the files and folders. I have made the new implementation match the spec as closely as possible, and left the existing ones the same size. This looks a bit weird right now, but as I mentioned I think it would be better to do away with having so many status indicators instead of sizing them differently.

### CODE REVIEW

File: src/powerhouse/components/dropdown-menu/dropdown-menu.tsx [hunk 1]
```diff
@@ -1,8 +1,8 @@
 import React from 'react';
 import {
     Button,
-    Item,
     Menu,
+    MenuItem,
```
Comment (ryanwolhuter) [inline src/powerhouse/components/dropdown-menu/dropdown-menu.tsx:5]: No idea why this happened. All of a sudden I got an error saying there is no `Item` exported here. Looked at the docs, and they do indeed use `MenuItem` and not `Item`.

**Comment by acaldas** [comment]
```
One thing I have noticed is that having these status indicators for all of the files and folders and drives makes the UI look a bit too busy. I think we need to consider showing less information. In the current designs, I notice that the status indicators for the individual files and folders are absent.
```

Yeah, in the previous designs there was an icon per node but this has changed now. Only the drives should have the status icon.

**Comment by acaldas** [comment]
https://github.com/powerhouse-inc/design-system/assets/5501872/bd003e12-6e07-4162-a9e3-42ab40b8c744

Do the tree vertical dots align with the icon in the design? We probably need to update the dots position

**Review by acaldas — APPROVED** [review]

**Review by gpuente — APPROVED** [review]
I didn't realize of the new designs, LGTM

**Pull request merged**